### PR TITLE
request-bottle: On CI, use merge commit author for co-author

### PR DESCRIPTION
**Pull request by issyl0**
- For auto-builds on GitHub Actions from merge commits, this used to set
  the co-authored-by line to the (Linux)BrewTestBot user.
- We do however want _some_ maintainer accountability for the actions of
  our bots, so set this to the author of the previous merge commit so
  there's a visible human involved.

### CODE REVIEW

File: cmd/request-bottle.rb [hunk 1]
```diff
@@ -17,21 +17,33 @@ def request_bottle_args
     end
   end
 
+  def head_is_merge_commit?
+    Utils.popen_read("git", "log", "--merges", "-1", "--format=%H").chomp == Utils.popen_read("git", "rev-parse", "HEAD").chomp
+  end
+
   def git_user
-    ENV["HOMEBREW_GIT_NAME"] || ENV["GIT_AUTHOR_NAME"] || ENV["GIT_COMMITTER_NAME"] || Utils.popen_read(Utils.git_path, "config", "--get", "user.name").strip
+    Utils.popen_read(Utils.git_path, "log", "-1", "--pretty=%an") if ENV['CI'] && head_is_merge_commit? \
```
Comment (issyl0) [inline cmd/request-bottle.rb]: I don't know if `ENV["CI"]` is a thing to find out if we're running on GitHub Actions or not.

Comment (dawidd6) [reply cmd/request-bottle.rb]: I think it is:
https://github.com/Homebrew/linuxbrew-core/commit/b3fff421ccc10e704611f2efb2403ba4a92388fe

File: cmd/request-bottle.rb [hunk 2]
```diff
@@ -17,21 +17,33 @@ def request_bottle_args
     end
   end
 
+  def head_is_merge_commit?
+    Utils.popen_read("git", "log", "--merges", "-1", "--format=%H").chomp == Utils.popen_read("git", "rev-parse", "HEAD").chomp
+  end
+
   def git_user
-    ENV["HOMEBREW_GIT_NAME"] || ENV["GIT_AUTHOR_NAME"] || ENV["GIT_COMMITTER_NAME"] || Utils.popen_read(Utils.git_path, "config", "--get", "user.name").strip
+    Utils.popen_read(Utils.git_path, "log", "-1", "--pretty=%an") if ENV["CI"] && head_is_merge_commit? \
```
Comment (jonchang) [inline cmd/request-bottle.rb]: I think I would prefer an early return (e.g., `return xyz if abc`) since this gets pretty hard to follow with the ifs and the ands and ors.

File: cmd/request-bottle.rb [hunk 3]
```diff
@@ -17,21 +17,33 @@ def request_bottle_args
...
+      || ENV["HOMEBREW_GIT_NAME"] \
+      || ENV["GIT_AUTHOR_NAME"] \
+      || ENV["GIT_COMMITTER_NAME"] \
+      || Utils.popen_read(Utils.git_path, "config", "--get", "user.name")
   end
 
   def git_email
-    ENV["HOMEBREW_GIT_EMAIL"] || ENV["GIT_AUTHOR_EMAIL"] || ENV["GIT_COMMITTER_EMAIL"] || Utils.popen_read(Utils.git_path, "config", "--get", "user.email").strip
+    Utils.popen_read(Utils.git_path, "log", "-1", "--pretty=%ae") if ENV["CI"] && head_is_merge_commit? \
```
Comment (jonchang) [inline cmd/request-bottle.rb]: Same as above.

**Comment by dawidd6** [comment]
Let's try it.
